craft_cli/messages.py

```diff
@@ -719,46 +721,63 @@ def message(self, text: str, intermediate: bool = False) -> None:
         also be used for important messages during the command's execution,
         with intermediate=True (which will include timestamp in verbose/trace mode).
         """
-        use_timestamp = bool(
-            intermediate and self._mode in (EmitterMode.VERBOSE, EmitterMode.TRACE)
-        )
-        self._printer.show(sys.stdout, text, use_timestamp=use_timestamp)  # type: ignore
+        # XXX Facundo 2022-06-23: the 'intermediate' modifier will be removed in next PRs
+        use_timestamp = bool(intermediate and self._mode in (EmitterMode.DEBUG, EmitterMode.TRACE))
+        stream = None if self._mode == EmitterMode.QUIET else sys.stdout
+        self._printer.show(stream, text, use_timestamp=use_timestamp)  # type: ignore
 
     @_active_guard()
     def trace(self, text: str) -> None:
-        """Trace/debug information.
+        """Trace information.
 
-        This is to record everything that the user may not want to normally see, but it's
-        useful for postmortem analysis.
-        """
-        stream = sys.stderr if self._mode == EmitterMode.TRACE else None
-        self._printer.show(stream, text, use_timestamp=True)  # type: ignore
-
-    @_active_guard()
-    def progress(self, text: str) -> None:
-        """Progress information for a multi-step command.
-
-        This is normally used to present several separated text messages.
-
-        These messages will be truncated to the terminal's width, and overwritten by the next
-        line (unless verbose/trace mode).
+        A way to expose system-generated information, about the general process or
+        particular information, which in general would be too overwhelming for
+        debugging purposes but sometimes needed for particular analysis.
         """
+        # if not in TRACE mode we're not even logging anything: so instead of calling the
```

Comment:
I improved the docstring to say that this method does not touch screen nor logs unless TRACE mode (which is a nice "API information") but left the big part of this comment as it is, because it's an internal implementation detail, not something to expose to the developer using the method.